azure/datalake/store/lib.py

```diff
@@ -86,6 +90,8 @@ def auth(tenant_id=default_tenant, username=default_username,
         the secret associated with the client_id
     resource : str
         resource for auth (e.g., https://management.core.windows.net/)
+    require_2fa : bool
+        indicates this authentication attempt requires two-factor authentication
```

Comment:
I agree with you. unfortunately we have to do some kind of wrapping in order to have reliable refresh token logic. If you can point me to a better pattern for handling refresh tokens and properly refreshing the token based only on the token, I am all for it.